OpenOversight/app/model_imports.py

```diff
@@ -50,9 +51,9 @@ def parse_int(value: Optional[Union[str, int]]) -> Optional[int]:
     return None
 
 
-def parse_float(value: Optional[Union[str, float]]) -> Optional[float]:
-    if value == 0.0 or value:
-        return float(value)
+def parse_decimal(value: Optional[Union[str, Decimal]]) -> Optional[Decimal]:
+    if value == Decimal(0) or value:
+        return Decimal(value)
     return None
```

Comment:
Is the goal of this function just to filter None? Other than None, all the potential inputs I can think of seem to be equivalent to just using the Decimal constructor:

| input | parse_decimal(input) | Decimal(input) |
|---|---|---|
| 0 | 0 | 0 |
| 1 | 1 | 1 |
| 1.1 | 1.1 | 1.1 |
| 0.0 | 0.0 | 0.0 |
| 0.01 | 0.01000000000000000020816681711721685132943093776702880859375 | 0.01000000000000000020816681711721685132943093776702880859375 |
| 0.0 | 0 | 0 |
| 0.0 | 0 | 0 |
| -1 | -1 | -1 |
| 0 | 0 | 0 |
| 1000.01 | 1000.009999999999990905052982270717620849609375 | 1000.009999999999990905052982270717620849609375 |

It seems like the function might be able to be simplified (from a reader's perspective) like this:

```suggestion
def parse_decimal(value: Optional[Union[str, Decimal]]) -> Optional[Decimal]:
    return Decimal(value) if value is not None else None
```

What do you think?

Reply:
I'm not familiar with this part of the code at all but the type hints seem to suggest that this function takes both string and decimal, so I think the original function might have been trying to filter out empty string as well as None. Does that make sense?

Reply:
Ahh, you're right, I totally missed that in my test. `Decimal("")` throws an error :+1: